interp/constrintern.ml

```diff
@@ -430,12 +416,12 @@ let locate_if_hole ?loc na c = match DAst.get c with
 
 let reset_hidden_inductive_implicit_test env =
   { env with impls = Id.Map.map (function
-         | (Inductive (params,_),b,c,d) -> (Inductive (params,false),b,c,d)
+         | (Inductive (params,_),b,c) -> (Inductive (params,false),b,c)
          | x -> x) env.impls }
 
 let check_hidden_implicit_parameters ?loc id impls =
   if Id.Map.exists (fun _ -> function
-    | (Inductive (indparams,check),_,_,_) when check -> Id.List.mem id indparams
+    | (Inductive (indparams,check),_,_) when check -> Id.List.mem id indparams
     | _ -> false) impls
   then
     user_err ?loc (Id.print id ++ strbrk " is already used as name of " ++
```

Comment:
This error is for instance triggered by 
~~~coq
Inductive foo {A:Prop} := C : forall A : Prop, foo.
~~~
which previously could not work because of the name manipulation removed by this PR.
Now it does work (assuming we remove the error). 
Removing the error then allows removing the arguments to `Inductive`.

Does that seem like something we should do? cc @herbelin

Reply:
I'm pushing a commit doing it, we can always drop it if we don't want it.

Reply:
What gives `Inductive foo {A:Prop} := C : forall A : Prop, foo.` with the PR? Is `A` renamed or is a failure telling that the name are not distinct?

In any case, if you have a way to avoid the "arguments to `Inductive`" hack, that's good.

Reply:
The second A is renamed to A0 AFAICT, ie
~~~coq
Inductive foo (A:Prop) := C : forall A : Prop, A -> foo _.
Print foo.
(* Inductive foo (A : Prop) : Type :=  C : forall A0 : Prop, A0 -> foo A *)
~~~

Reply:
> The second A is renamed to A0 AFAICT, ie

I think, it is ok for this PR. In the plan I have around implicit arguments, there would be anyway a check that the names are distinct in the toplevel block of parameters.